bindings/wasm/output_test.go

```diff
@@ -28,11 +28,15 @@ import (
 )
 
 const (
-	pathArgs    = "./testdata/args/main.wasm"
-	pathExample = "./testdata/example/main.wasm"
-	pathLoop    = "./testdata/loop/main.wasm"
+	urlArgsFile    = "file://testdata/args/main.wasm"
+	urlExampleFile = "file://testdata/example/main.wasm"
+	urlLoopFile    = "file://testdata/loop/main.wasm"
+	urlPythonOCI   = "oci://ghcr.io/vmware-labs/python-wasm:3.11.3"
```

Comment:
@codefromthecrypt is this an example? is that a python script? or some kind of python runtime?

Reply:
it is actually the python runtime. So there are basically two main ways to do wasm.

1. compile your source to wasm (c, go, etc.)
2. compile your interpreter to wasm (python, ruby, etc.)

So, in this case, the python script could be taken from stdin or args. For example, you could do an inline script in args, or if someone added a layer with a script, you could indicate that.
```
-c "import sys; import pprint; pprint.pprint(sys.path)"
```

The above are the ARGS used in the README of that image.
https://github.com/vmware-labs/webassembly-language-runtimes/blob/c6cf6e11ffaef1624108dfc9581911ed11f3f42d/python/README.md?plain=1#L72

Interpreters often need standard libraries mounted in some way, and this is where OCI comes in. There are layers for the root filesystem that also includes supporting files. This prevents users from having to deal with this or potentially add the wrong versions of these files (or mess up variables. For example, the OCI image linked has the correct PYTHONPATH/PYTHONHOME hard-coded into python.wasm and the files are in the image layers:

```bash
$  ./car -tvvf ghcr.io/vmware-labs/python-wasm:3.11.3
https://ghcr.io/v2/vmware-labs/python-wasm/manifests/3.11.3 platform=wasi/wasm32 totalLayerSize: 9584425
https://ghcr.io/v2/vmware-labs/python-wasm/blobs/sha256:2940f0fe18fb54307399ab895bdfaa2a45d13bec713bef86d07f205e9b223552 size=5516755
CreatedBy: COPY python/v3.11.3/bin/python.wasm /python.wasm # buildkit
-rw-r--r--	14756882	Apr 26 19:13:12	python.wasm
https://ghcr.io/v2/vmware-labs/python-wasm/blobs/sha256:8b1e220740c34229cc74ece6224070d2ca8f5d797b0a7d42dc62809575b2bf7f size=4067670
CreatedBy: COPY python/v3.11.3/usr/ /usr # buildkit
-rw-r--r--	0	Apr 26 19:13:12	usr/local/lib/python3.11/lib-dynload/.empty
-rw-r--r--	39504	Apr 26 19:13:12	usr/local/lib/python3.11/os.py
-rw-r--r--	4097136	Apr 26 19:13:12	usr/local/lib/python311.zip
```

Long answer, but hope it helps share the knowledge I accidentally have ;)

Reply:
PS I chose python for OCI because it uses the most features of OCI. A simple case OCI image will have only one layer, the wasm file. Interpreters will need to read the script from somewhere, so only really inlined stuff will be able to use simple case of a wasm file.

Reply:
hopefully clarified in https://github.com/dapr/docs/pull/3362